include/envoy/http/header_map.h

```diff
@@ -22,6 +22,10 @@ class LowerCaseString {
 
   const std::string& get() const { return string_; }
   bool operator==(const LowerCaseString& rhs) const { return string_ == rhs.string_; }
+  LowerCaseString& operator=(const LowerCaseString& rhs) {
```

Comment:
as far as i can tell the following happens:
Optional.value(arg) method does copy assignment for the argument passed in, what ends up happening vector<LowerCaseString> is copy assigned and consequently LowerCaseString is copy assigned which is not permitted as we have move ctor for the LowerCaseString and compiler removes implicit operator=(const LowerCaseString&).

That's why one solution is to explicitly define copy assignment.